dataland-frontend/tests/e2e/utils/EuTaxonomyFinancialsUpload.ts

```diff
@@ -304,4 +320,57 @@ export function fillEuTaxonomyForFinancialsRequiredFields(data: EuTaxonomyDataFo
   cy.get('[data-test="assuranceSection"] select[name="assurance"]').select(2);
   cy.get('[data-test="assuranceSection"] input[name="provider"]').type("Assurance Provider", { force: true });
   cy.get('[data-test="assuranceSection"] select[name="report"]').select(1);
+
+  cy.get('[data-test="MultiSelectfinancialServicesTypes"]')
+    .click()
+    .get("div.p-multiselect-panel")
+    .find("li.p-multiselect-item")
+    .each(($el) => {
+      cy.wrap($el).click({ force: true });
+    });
+
+  cy.get('[data-test="addKpisButton"]').click({ force: true });
+
+  cy.get('[data-test="removeSectionButton"]').each(($el, index) => {
+    if (index > 0) {
+      cy.wrap($el).click({ force: true });
+    }
+  });
+
+  cy.get('button[data-test="removeSectionButton"]').should("exist").should("have.class", "ml-auto");
+
+  fillEligibilityKpis("creditInstitutionKpis", data.eligibilityKpis?.CreditInstitution);
+  fillField(
+    "creditInstitutionKpis",
+    "tradingPortfolioAndInterbankLoans",
+    data.creditInstitutionKpis?.tradingPortfolioAndInterbankLoans
+  );
+  fillField("creditInstitutionKpis", "tradingPortfolio", data.creditInstitutionKpis?.tradingPortfolio);
+  fillField("creditInstitutionKpis", "interbankLoans", data.creditInstitutionKpis?.interbankLoans);
+  fillField("creditInstitutionKpis", "greenAssetRatio", data.creditInstitutionKpis?.greenAssetRatio);
+}
+
+/**
+ * Fills the eutaxonomy-financials upload form with the given dataset
+ * @param data the data to fill the form with
+ * @param reportingPeriod (optional) to specify reporting period
+ * @param submissionDataIntercept function that asserts content of an intercepted request
+ */
+export function addCreditInstitutionDataset(
+  data: EuTaxonomyDataForFinancials,
+  reportingPeriod?: string,
+  submissionDataIntercept?: (request: CyHttpMessages.IncomingHttpRequest) => void
+): void {
+  fillAndValidateEuTaxonomyCreditInstitutionForm(data, reportingPeriod);
+  submitFilledInEuTaxonomyForm(submissionDataIntercept);
+}
+
+/**
+ * @param fixtureType name of fixture data to prepare
+ * @returns cypress chainable fixture containing array of fixture data
+ */
+export function prepareFixture<T>(fixtureType: string): Chainable<FixtureData<T>[]> {
+  return cy.fixture(fixtureType).then(function (jsonContent) {
+    return jsonContent as Array<FixtureData<T>>;
+  });
```

Comment:
It's ok for me that you put this into a method, because it is done again and again in many tests.
But:
1.) I think it should not be in this file `EuTaxonomyFinancialsUpload.ts`, because it is not specific to the eu taxonomy financials framework. It should rather be in a more general utils file.
2.) If you want to introduce this, we should also use it consistently everywhere.
There are still a lot of occurences of `cy.fixture(...)` in our test code where one could use this method now.

Reply:
Yeah I saw that. There is **a lot** of uses. 49 to be exact. Should I update them as part of this??
Moved it to `GeneralApiUtils.ts`.

Reply:
Actually, a lot of test generate the fixtures differently. Per `it()`. Perhaps I'll revert my code and remove this function.

Reply:
Ah ok that's really a lot.
Both of your suggestions are ok to me.   So either revert your change, or do it everywhere else.

But I suggest you create a new file `GeneralUtils.ts` for it.
The `GeneralApiUtils.ts` was meant to contain only generally needed util functions that do request to our backend.

Reply:
👍